### PR TITLE
fix: fix release workflow

### PR DESCRIPTION
**Pull request by mkuznyetsov**
Signed-off-by: Mykhailo Kuznietsov <mkuznets@redhat.com>

Fix Che Code release workflow errors, that were discovered after debugging the action live

### CODE REVIEW

**Review by azatsarynnyy — APPROVED** [review]

File: .github/workflows/image-publish.yml [hunk 1]
```diff
@@ -83,6 +83,10 @@ jobs:
     # don't notify for cancelled builds
     if: (success() || failure()) && github.ref != 'refs/heads/main'
     steps:
+      - name: Set variables
+        run: |
+          BRANCH_NAME=${{ github.ref }}
+          echo "BRANCH_NAME=${BRANCH_NAME##*/}" >> $GITHUB_ENV
```
Comment (nickboldt) [inline .github/workflows/image-publish.yml:89]: On line 93, you're pointing to the other action, not this one.

File: make-release.sh [hunk 2]
```diff
@@ -126,6 +130,7 @@ fi
 # now update ${BASEBRANCH} to the new version
 git checkout "${BASEBRANCH}"
 
+
```
Comment (nickboldt) [inline make-release.sh]: extra line?

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -57,5 +59,17 @@ jobs:
           git config --global user.name "Mykhailo Kuznietsov"
           git config --global user.email "mkuznets@redhat.com"
           git config --global pull.rebase true
-          export GITHUB_TOKEN=${{ secrets.CHE_INCUBATOR_BOT_GITHUB_TOKEN }}
+          export GITHUB_TOKEN=${{ secrets.CHE_INCUBATOR_BOT_TOKEN }}
           /bin/bash make-release.sh --version ${{ github.event.inputs.version }} --tag-release
+      - name: Create failure MM message
+        if: failure()
+        run: |
+          echo "{\"text\":\":no_entry_sign: Che Code ${{ env.BRANCH_NAME }} release has failed: https://github.com/che-incubator/che-code/actions/workflows/release.yml\"}" > mattermost.json
```
Comment (nickboldt) [inline .github/workflows/release.yml:67]:
```suggestion
          echo "{\"text\":\":no_entry_sign: Che Code ${{ env.BRANCH_NAME }} release has failed: https://github.com/che-incubator/che-code/actions/workflows/image-publish.yml\"}" > mattermost.json
```

File: make-release.sh (hunk 2 above)
Comment (nickboldt) [inline make-release.sh]:
```suggestion
```

File: make-release.sh [hunk 4]
```diff
@@ -43,13 +43,17 @@ bump_version () {
   BUMP_BRANCH=$2
 
   git checkout "${BUMP_BRANCH}"
-
+  if [[ $(grep version package.json | awk -F \" '{print $4}') == ${NEXT_VERSION} ]]; then
```
Comment (nickboldt) [inline make-release.sh:46]: you could do this in one line with jq, but grep|awk works too, I suppose.

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -57,5 +59,17 @@ jobs:
           git config --global user.name "Mykhailo Kuznietsov"
           git config --global user.email "mkuznets@redhat.com"
           git config --global pull.rebase true
-          export GITHUB_TOKEN=${{ secrets.CHE_INCUBATOR_BOT_GITHUB_TOKEN }}
+          export GITHUB_TOKEN=${{ secrets.CHE_INCUBATOR_BOT_TOKEN }}
           /bin/bash make-release.sh --version ${{ github.event.inputs.version }} --tag-release
+      - name: Create failure MM message
+        if: failure()
+        run: |
+          echo "{\"text\":\":no_entry_sign: Che Code ${{ env.BRANCH_NAME }} release has failed: https://github.com/che-incubator/che-code/actions/workflows/image-publish.yml\"}" > mattermost.json
```
Comment (nickboldt) [inline .github/workflows/release.yml]:
```suggestion
          echo "{\"text\":\":no_entry_sign: Che Code ${{ env.BRANCH_NAME }} release has failed: https://github.com/che-incubator/che-code/actions/workflows/release.yml\"}" > mattermost.json
```

**Review by nickboldt — APPROVED** [review]

**Pull request merged**